src/Agent.Plugins/PipelineArtifact/FileShareProvider.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Agent.Sdk;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.BlobStore.Common;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Plugins.PipelineArtifact
+{
+    internal class FileShareProvider: IArtifactProvider
+    {
```

Comment:
This is something that we can write unit tests on the agent side repo easily because it doesn't have a dependency on any service calls, can you write some unit tests that copy from one dir to another dir? It doesn't need to be from a file share as copying from a file share is really just copying from one part of the local file system to another.

Reply:
On windows there is an implicit admin file share:
```
\\localhost\[drive letter]$\
```
e.g.
```
C:\Users\jerick>dir \\localhost\c$
 Volume in drive \\localhost\c$ is OSDisk
 Volume Serial Number is 10A1-4A29

 Directory of \\localhost\c$

06/20/2019  04:38 PM    <DIR>          debuggers64
06/03/2019  09:40 AM    <DIR>          Deploy
06/03/2019  09:31 AM    <DIR>          Intel
06/21/2019  08:19 AM    <DIR>          LR
08/02/2019  12:33 AM    <DIR>          Microsoft
```